alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java

```diff
@@ -783,27 +783,16 @@ private void actionTXREPORT(SolrQueryResponse rsp, SolrParams params, String cna
         {
             throw new AlfrescoRuntimeException("No txid parameter set");
         }
-
-        if (cname != null)
+        if (cname == null)
         {
-            MetadataTracker tracker = trackerRegistry.getTrackerForCore(cname, MetadataTracker.class);
-            Long txid = Long.valueOf(params.get(ARG_TXID));
-            NamedList<Object> report = new SimpleOrderedMap<Object>();
-            report.add(cname, buildTxReport(getTrackerRegistry(), informationServers.get(cname), cname, tracker, txid));
-            rsp.add("report", report);
-        }
-        else
-        {
-            Long txid = Long.valueOf(params.get(ARG_TXID));
-            NamedList<Object> report = new SimpleOrderedMap<Object>();
-            for (String coreName : trackerRegistry.getCoreNames())

```

Comment:
The only way this might not throw an exception is if there are no cores in the registry.  I've assumed that this won't happen, but it's worth reviewing.

Reply:
This would change the behaviour of the report, not having cname would return a report for all the cores acl txns. The proposed change will now throw "No cname parameter set"

Reply:
Not having cname set would previously throw a NullPointerException in trackerRegistry.getTrackerForCore (which will then be wrapped by a SolrException). With the new code we will get an AlfrescoRuntimeException wrapped by a SolrException.

The only edge case is that previously this method would do nothing if there are no cores in the registry and null is passed in. Now it will throw an exception.

I'll add some unit tests and check them before and after my changes.  I don't plan to add a unit test for the edge case I've described as I don't think it's a reasonable use of the transaction reporting.